src/EnergyPlus/HybridModel.cc

```diff
@@ -0,0 +1,163 @@
+	
+	bool FlagHybridModel( false ); // True if hybrid model is activated
+	bool FlagHybridModelInf( false ); // True if hybrid model (infiltration) is activated
+	int NumOfHybridModelZones( 0 ); // Number of hybrid model zones in the model
+	std::string CurrentModuleObject; // to assist in getting input
+
+	// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+
+	// Object Data
+	Array1D< HybridModelProperties > HybridModelZone;
+
+	// Functions
+
+	void
+	CheckAndReadHybridModelZone()
+	{
+
+		using ScheduleManager::GetScheduleIndex;
+
+		static bool RunMeOnceFlag( false );
+		static bool ErrorsFound( false ); // If errors detected in input
+		static Array1D_bool lAlphaFieldBlanks( 10, false );
+		static Array1D_bool lNumericFieldBlanks( 10, false );
+		int NumAlphas; // Number of Alphas for each GetobjectItem call
+		int NumNumbers; // Number of Numbers for each GetobjectItem call
+		int IOStatus;
+		int ZonePtr; // Pointer to the zone
+		int ZoneListPtr; // Pointer to the zone list
+		std::string CurrentModuleObject; // to assist in getting input
+		Array1D_string cAlphaArgs( 10 ); // Alpha input items for object
+		Array1D_string cAlphaFieldNames( 10 );
+		Array1D_string cNumericFieldNames( 10 );
+		Array1D< Real64 > rNumericArgs( 10 ); // Numeric input items for object
+
+		if ( RunMeOnceFlag ) return;
+
+		// Read hybrid model input
+		CurrentModuleObject = "HybridModel:Zone";
+		NumOfHybridModelZones = GetNumObjectsFound( CurrentModuleObject );
+		HybridModelZone.allocate( NumOfZones );
+
+		if ( NumOfHybridModelZones > 0 ) {
+
+			RunMeOnceFlag = true;
+		
+			for ( int HybridModelNum = 1; HybridModelNum <= NumOfHybridModelZones; ++HybridModelNum ) {
+
+				GetObjectItem( CurrentModuleObject, HybridModelNum, cAlphaArgs, NumAlphas, rNumericArgs, NumNumbers, IOStatus, lNumericFieldBlanks, lAlphaFieldBlanks, cAlphaFieldNames, cNumericFieldNames );
+				
+				ZoneListPtr = 0;
+				ZonePtr = FindItemInList( cAlphaArgs( 2 ), Zone );
+				if ( ZonePtr == 0 && NumOfZoneLists > 0 ) ZoneListPtr = FindItemInList( cAlphaArgs( 2 ), ZoneList );
+				if ( ZonePtr > 0 ) {
+					HybridModelZone( ZonePtr ).Name = cAlphaArgs( 1 );
+					HybridModelZone( ZonePtr ).InternalThermalMassCalc = SameString( cAlphaArgs( 3 ), "YES" );
+					HybridModelZone( ZonePtr ).InfiltrationCalc = SameString( cAlphaArgs( 4 ), "YES" );
+					HybridModelZone( ZonePtr ).ZoneMeasuredTemperatureSchedulePtr = GetScheduleIndex( cAlphaArgs( 5 ) );
+					HybridModelZone( ZonePtr ).ZoneMeasuredTemperatureStartMonth = rNumericArgs( 1 );
+					HybridModelZone( ZonePtr ).ZoneMeasuredTemperatureStartDate = rNumericArgs( 2 );
+					HybridModelZone( ZonePtr ).ZoneMeasuredTemperatureEndMonth = rNumericArgs( 3 );
+					HybridModelZone( ZonePtr ).ZoneMeasuredTemperatureEndDate = rNumericArgs( 4 );
+				} else if ( ZoneListPtr > 0 ) {
+					for( int ZonePtrNum = 1; ZonePtrNum < ZoneList( ZoneListPtr ).NumOfZones; ZonePtrNum++ ){
+						HybridModelZone( ZonePtr ).Name = cAlphaArgs( 1 );
```

Comment:
@sanghlee-lbnl When NumOfZoneLists> 0,  ZonePtr = 0. Therefore, ZonePtr should not be used as index.